mergify_engine_hooks/datadog_hook/__init__.py

```diff
@@ -0,0 +1,26 @@
+# -*- encoding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from datadog import statsd
+
+from mergify_engine import github_types
+from mergify_engine import hooks
+
+
+class Hook(hooks.HookBase):
+    async def __call__(
+        self, account: github_types.GitHubAccount, event: hooks.EventName
+    ) -> None:
+        if event.startswith("action"):
+            statsd.increment("engine.hooks.action.count", tags=[f"event:{event[7:]}"])
```

Comment:
1. This looks different than what we currently have. Is this better/necessary? Should we only have one metric?
2. I'd remove `hooks` from the name. The mechanism used to gather the metric is not useful in the name of the metric I'd say 🤔

Reply:
I describe in the commit message what we are currently tracking and what this track, this is not the same. We need both I would say. Customer doesn't need to implementation details of the engine. While we want it for the monitoring.